system/jhtree/jhinplace.cpp

```diff
@@ -2109,13 +2127,21 @@ bool CInplaceLeafWriteNode::add(offset_t pos, const void * _data, size32_t size,
                         //MORE: Some compressors do not support incremental compression.  For those we should continue to
                         //gather uncompressed, but may also need to move the buffer to the context so it is reused (or
                         //maybe swap with the context).
-                        gatherUncompressed = false;
+                        if (!ctx.options.recompress)
+                            gatherUncompressed = false;
                     }
                 }
                 // else compressed does not fit either - better to remain uncompressed.
             }
             else
             {
+                if (ctx.options.recompress)// && uncompressed.length() < nodeSize * 10)
+                {
+                    //Try recompressing all the payload and see if there is enough room after that.
+                    if (recompressAll(maxPayloadSize))
```

Comment:
Not sure I have 100% followed the logic, but is this continuously recompressing the entire page (as it starts getting close to being full) and if so is there a performance implication?

Reply:
Yes it is.  The impact isn't as high as I expected - and the performance seems adequate even on highly compressible payloads like codesv3.